src/couch/src/couch_db.erl

```diff
@@ -945,13 +945,14 @@ prep_and_validate_replicated_updates(Db, [Bucket|RestBuckets], [OldInfo|RestOldI
     {ok, #full_doc_info{rev_tree=OldTree}} ->
         OldLeafs = couch_key_tree:get_all_leafs_full(OldTree),
         OldLeafsLU = [{Start, RevId} || {Start, [{RevId, _}|_]} <- OldLeafs],
+        NewPaths = lists:map(fun couch_doc:to_path/1, Bucket),
+        SortedPaths = lists:sort(NewPaths),
         NewRevTree = lists:foldl(
-            fun(NewDoc, AccTree) ->
-                {NewTree, _} = couch_key_tree:merge(AccTree,
-                    couch_doc:to_path(NewDoc), Db#db.revs_limit),
+            fun(NewPath, AccTree) ->
+                {NewTree, _} = couch_key_tree:merge(AccTree, NewPath),
```

Comment:
No, I'm asserting stemming isn't needed here because we don't use the output. In this particular instances we throw away the result of the stem regardless.

I *am* relying on the fact that the new merge doesn't produce a broken rev tree though. Here all we're doing is looking for new leaves that have to be passed through the VDU. The stem isn't actually necessary because NewRevTree is thrown away after we discover the list of new leaves that need validated.